File: src/main/kotlin/com/itmo/microservices/demo/order/impl/service/IOrderService.java
package com.itmo.microservices.demo.order.impl.service;

import java.util.UUID;

public interface IOrderService {
    Order createOrder(Order order);
    Order getOrderById(UUID orderId);
    void updateOrder (UUID orderId, UUID itemId, int amount);
    Booking book(UUID orderId);
    void selectDeliveryTime(UUID orderId, int seconds);
}

File: src/main/kotlin/com/itmo/microservices/demo/order/impl/service/IOrderService.kt
package com.itmo.microservices.demo.order.impl.service

import com.itmo.microservices.demo.order.api.dto.BookingDto
import com.itmo.microservices.demo.order.api.dto.OrderDto
import java.util.*

interface IOrderService {
    fun selectDeliveryTime(orderId: UUID?, seconds: Int)
    fun book(orderId: UUID?): BookingDto?
    fun putItemToOrder(orderId: UUID?, itemId: UUID?, amount: Int)
    fun getOrderById(uuid: UUID?): OrderDto?
    fun createOrder(): OrderDto?
}